### PR TITLE
Backport "lxd init" code structure changes 

### PR DESCRIPTION
**Pull request by freeekanayaka**

### CODE REVIEW

File: lxd/main_init.go [hunk 1]
```diff
@@ -273,44 +281,256 @@ they otherwise would.
...
+// revert it to its original state.
+func (cmd *CmdInit) initPoolUpdate(client lxd.ContainerServer, pool api.StoragePoolsPost, currentPool api.StoragePoolPut) (reverter, error) {
+	reverter := func() error {
+		return client.UpdateStoragePool(pool.Name, currentPool, "")
+	}
+	err := client.UpdateStoragePool(pool.Name, api.StoragePoolPut{
+		Config: pool.Config,
+	}, "")
+	return reverter, err
+}
```
Comment (stgraber) [inline lxd/main_init.go]: The storage pool API doesn't exist in LXD 2.0, so I don't see how initPoolCreate and initPoolUpdate can possibly work.

File: lxd/main_init.go [hunk 2]
```diff
@@ -273,44 +281,256 @@ they otherwise would.
...
+// revert it to its original state.
+func (cmd *CmdInit) initNetworkUpdate(client lxd.ContainerServer, network api.NetworksPost, currentNetwork api.NetworkPut) (reverter, error) {
+	reverter := func() error {
+		return client.UpdateNetwork(network.Name, currentNetwork, "")
+	}
+	err := client.UpdateNetwork(network.Name, api.NetworkPut{
+		Config: network.Config,
+	}, "")
+	return reverter, err
+}
```
Comment (stgraber) [inline lxd/main_init.go]: Same comment as with storage, the network API doesn't exist in LXD 2.0.

Comment (freeekanayaka) [reply lxd/main_init.go]: Same as above.

File: lxd/main_init.go (hunk 1 above)
Comment (freeekanayaka) [reply lxd/main_init.go]: Good catch, this is artifact of the cherry pick, but effectively idle code that can't be reached. Removed.

File: lxd/main_init.go (hunk 2 above)
Comment (stgraber) [reply lxd/main_init.go]: Did you forget to remove that code too?

Comment (freeekanayaka) [reply lxd/main_init.go]: Ah, a "don't do try to do that late at night commit". Dropped for real.

**Pull request merged**